indico/modules/rb/tasks.py

```diff
@@ -111,9 +62,17 @@ def roombooking_occurrences():
 
     try:
         for occ in occurrences:
-            notify_upcoming_occurrence(occ)
-            occ.notification_sent = True
-            if occ.reservation.repeat_frequency == RepeatFrequency.DAY:
-                occ.reservation.occurrences.update({'notification_sent': True})
+            same_user_occurrences = []
+            for other in occurrences:
```

Comment:
I don't think this is the best/more efficient way of doing what you want to do. Have you thought of a group by query?